### PR TITLE
feat(kubernetes_platform): Add empty dir mount

### PR DESCRIPTION
**Pull request by gregsheremeta**
**Description of your changes:**
Update kubernetes_platform go package to include the ability to add an EmptyDir mount. 

Part 1 of 2. Will be rebased when the backend part is merged. Posting as a WIP now to get feedback on the fields in the proto message.

Following the example set in #10410.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by gregsheremeta** [comment]
@chensun do you know whom I can nag about reviewing kfp-kubernetes PRs? I ask because it seems like Connor left the project.

File: kubernetes_platform/proto/kubernetes_executor_config.proto [hunk 1]
```diff
@@ -38,6 +38,7 @@ message KubernetesExecutorConfig {
     repeated NodeAffinityTerm node_affinity = 14;
     repeated PodAffinityTerm pod_affinity = 15;
     EnabledSharedMemory enabled_shared_memory = 16;
+    repeated EmptyDirMount empty_dir_mount = 17;
```
Comment (chensun) [inline kubernetes_platform/proto/kubernetes_executor_config.proto:41]: nit: empty_dir_mounts

**Review by chensun — COMMENTED** [review]
/lgtm

File: kubernetes_platform/proto/kubernetes_executor_config.proto (hunk 1 above)
Comment (gregsheremeta) [reply kubernetes_platform/proto/kubernetes_executor_config.proto:41]: interesting. I don't see plurals anywhere else -- for example, line 26:
```
repeated PvcMount pvc_mount = 3;
```

that said, this is my very first time doing anything with protobuf, so if repeated fields should be pluralized, I will pluralize it.

**Comment by gregsheremeta** [comment]
s/WIP//

I think it's good for final review

**Review by hbelmiro — COMMENTED** [review]
/lgtm

File: kubernetes_platform/proto/kubernetes_executor_config.proto (hunk 1 above)
Comment (gregsheremeta) [reply kubernetes_platform/proto/kubernetes_executor_config.proto]: ref: https://cloud.google.com/apis/design/naming_convention#repeated_field_names

> Repeated fields in APIs must use proper plural forms. This matches the convention of existing Google APIs, and the common expectation of external developers.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**